src/ir.rs

```diff
@@ -335,10 +370,28 @@ pub struct Program
 	//pub shader_modules : HashMap<usize, ShaderModule>
 }
 
+impl serde::Serialize for Program {
```

Comment:
I don't think we need this now with Mateo's improved StableVec type

Reply:
I was too scared to change the hashmaps and do the refactoring :/

Part of it is that you would lose the type indication of "FuncletId", which I didn't like, but I can change it